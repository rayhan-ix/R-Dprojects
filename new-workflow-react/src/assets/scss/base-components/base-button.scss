$base-btn-minWidth-without-icon: 100px;
$base-btn-minHeight-without-icon: 30px;

$base-btn-minWidth-with-icon: 110px;
$base-btn-minHeight-with-icon: 35px;

$base-btn-minWidth: 100px;
$base-btn-minHeight: 30px;

$base-icon-btn-minWidth: 20px;
$base-icon-btn-minHeight: 20px;

.base-btn {
  border-radius: 5px;
  min-width: $base-btn-minWidth;
  min-height: $base-btn-minHeight;
}

.base-btn-without-icon {
  @extend .base-btn;
  min-width: $base-btn-minWidth-without-icon;
  min-height: $base-btn-minHeight-without-icon;
}

.base-btn-with-icon {
  @extend .base-btn;
  min-width: $base-btn-minWidth-with-icon;
  min-height: $base-btn-minHeight-with-icon;
}

.base-icon-btn {
  @extend .base-btn;
  min-width: $base-icon-btn-minWidth;
  min-height: $base-icon-btn-minHeight;
}
