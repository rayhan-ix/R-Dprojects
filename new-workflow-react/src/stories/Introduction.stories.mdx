import {Meta} from "@storybook/addon-docs/blocks";
import Repo from "./assets/repo.svg";

<Meta title="Introduction" />

<style>{`
  .subheading {
    --mediumdark: '#999999';
    font-weight: 900;
    font-size: 13px;
    color: #999;
    letter-spacing: 6px;
    line-height: 24px;
    text-transform: uppercase;
    margin-bottom: 12px;
    margin-top: 40px;
  }

  .link-list {
    display: grid;
    grid-template-columns: 1fr;
    grid-template-rows: 1fr 1fr;
    row-gap: 10px;
  }

  @media (min-width: 620px) {
    .link-list {
      row-gap: 20px;
      column-gap: 20px;
      grid-template-columns: 1fr 1fr;
    }
  }

  @media all and (-ms-high-contrast:none) {
  .link-list {
      display: -ms-grid;
      -ms-grid-columns: 1fr 1fr;
      -ms-grid-rows: 1fr 1fr;
    }
  }

  .link-item {
    display: block;
    padding: 20px 30px 20px 15px;
    border: 1px solid #00000010;
    border-radius: 5px;
    transition: background 150ms ease-out, border 150ms ease-out, transform 150ms ease-out;
    color: #333333;
    display: flex;
    align-items: flex-start;
  }

  .link-item:hover {
    border-color: #1EA7FD50;
    transform: translate3d(0, -3px, 0);
    box-shadow: rgba(0, 0, 0, 0.08) 0 3px 10px 0;
  }

  .link-item:active {
    border-color: #1EA7FD;
    transform: translate3d(0, 0, 0);
  }

  .link-item strong {
    font-weight: 700;
    display: block;
    margin-bottom: 2px;
  }

  .link-item img {
    height: 40px;
    width: 40px;
    margin-right: 15px;
    flex: none;
  }

  .link-item span {
    font-size: 14px;
    line-height: 20px;
  }

  .link-item em {
    font-size: 15px;
    font-weight: bold;
  }

  .tip {
    display: inline-block;
    border-radius: 1em;
    font-size: 11px;
    line-height: 12px;
    font-weight: 700;
    background: #E7FDD8;
    color: #66BF3C;
    padding: 4px 12px;
    margin-right: 10px;
    vertical-align: top;
  }

  .tip-wrapper {
    font-size: 13px;
    line-height: 20px;
    margin-top: 40px;
    margin-bottom: 40px;
  }

  .tip-wrapper code {
    font-size: 12px;
    display: inline-block;
  }


`}</style>

# Welcome to Storybook of New Workflow App

Storybook helps you build UI components in isolation from your app's business logic, data, and context.
That makes it easy to develop hard-to-reach states. Save these UI states as **stories** to revisit during development, testing, or QA.

Browse stories by navigating to them in the sidebar.
Storybook recommend building UIs with a [**component-driven**](https://componentdriven.org) process starting with atomic components and ending with pages.

<div className="subheading">Directory Structure</div>

<div className="link-list">
    <div className="link-item">
    <span>
      <strong>assets</strong>
      Contains <em>scss file </em> and <em>images</em> and <em>icons</em>.
    </span>
    </div>
    <div className="link-item">
    <span>
      <strong>components</strong>
      Contains all the base components which can be imported in other <em>
        pages and page components.
      </em>
    </span>
    </div>
    <div className="link-item">
    <span>
      <strong>modules</strong>
      Contains three modules <em>
        users, system-admin, operation-manager
      </em> which includes <em>pages</em> defined in route.
    </span>
    </div>
    <div className="link-item">
    <span>
      <strong>store</strong>
      Redux store for the <em>modules</em>
    </span>
    </div>
    <div className="link-item">
    <span>
      <strong>stories</strong>
      stories for each <em>components</em> and <em>pages</em>
    </span>
    </div>
    <div className="link-item">
    <span>
      <strong>utils</strong>
      For <em>utility</em> and <em>helper</em> functions
    </span>
    </div>
</div>

<div className="subheading">Technology Used</div>

<div className="link-list">
    <a className="link-item" href="https://formik.org/" target="_blank">
        <img src={Repo} alt="repo" />
        <span>
      <strong>Formik</strong>
      For building forms
    </span>
    </a>
    <a className="link-item" href="https://ant.design/" target="_blank">
        <img src={Repo} alt="repo" />
        <span>
      <strong>Ant Design</strong>
      Design Framework
    </span>
    </a>
    <a
        className="link-item"
        href="https://github.com/alibaba/react-intl-universal"
        target="_blank"
    >
        <img src={Repo} alt="repo" />
        <span>
      <strong>React Internationalization</strong>
      Language support
    </span>
    </a>
    <a
        className="link-item"
        href="https://github.com/jquense/yup"
        target="_blank"
    >
        <img src={Repo} alt="repo" />
        <span>
      <strong>Yup</strong>
      Schema validator
    </span>
    </a>
    <a className="link-item" href="https://formbuilder.online/" target="_blank">
        <img src={Repo} alt="repo" />
        <span>
      <strong>Drag and Drop form</strong>
      Form Builder
    </span>
    </a>
    <a className="link-item" href="https://storybook.js.org/" target="_blank">
        <img src={Repo} alt="repo" />
        <span>
      <strong>Storybook</strong>
      For ui documentation
    </span>
    </a>
    <a
        className="link-item"
        href="https://www.typescriptlang.org/"
        target="_blank"
    >
        <img src={Repo} alt="repo" />
        <span>
      <strong>Typescript</strong>
    </span>
    </a>
</div>

<div className="subheading">NB</div>

> - #### If possible, we can use [**presentation-container component**](https://dev.to/olenadrugalya/presentational-and-container-components-in-react-4gfj) pattern

> - #### If possible, we can keep updating the stories/documentation of ui here

<div className="tip-wrapper">
    <span className="tip">Tip</span>Edit the Markdown in{" "}
    <code>src/stories/Introduction.stories.mdx</code>
</div>
