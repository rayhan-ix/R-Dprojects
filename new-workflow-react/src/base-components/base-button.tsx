/* eslint-disable no-unused-vars */
import React, { ReactNode } from 'react';
import { Button } from 'antd';
import '../assets/scss/base-components/base-button.scss';
import clsx from 'clsx';

export interface BaseButtonProps {
  /**
   * Button Label. example: Save, Cancel, etc
   * Pass label null if it is only an icon button
   * Default Button
   */
  label?: string;

  /**
   * Icon to show at the suffix of the button or to show only icon button provided label is null
   * Default null
   */
  icon?: ReactNode | null;

  /**
   * If the button should be disabled
   * Default false
   */
  disabled?: boolean;

  /**
   * html button type
   *
   */
  buttonType: 'button' | 'reset' | 'submit';

  /**
   * onClick handler for
   * @param args
   */
  onClick: (args?: any) => any;
}

export const BaseButton = (props: BaseButtonProps) => {
  const { label, buttonType, disabled = false, icon = null, onClick } = props;

  return (
    <Button
      type='primary'
      className={clsx([
        { 'base-btn-without-icon': !icon },
        { 'base-btn-with-icon': !!icon && label },
        { 'base-icon-btn': !!icon && !label },
      ])}
      htmlType={buttonType}
      disabled={disabled}
      icon={icon}
      onClick={onClick}
    >
      {label}
    </Button>
  );
};
