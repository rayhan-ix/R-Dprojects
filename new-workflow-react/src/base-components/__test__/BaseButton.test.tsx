import React from 'react';
import { render, screen } from '@testing-library/react';
import { BaseButton } from '../base-button';
import { IconButton, WithIcon } from '../../stories/base-components/BaseButton.stories';

describe('Button Testing', () => {
  it('should render normal button', () => {
    render(<BaseButton label='Primary' buttonType='button' onClick={() => jest.fn()} />);
    const button = screen.getByRole('button');
    expect(button).toHaveTextContent('Primary');
  });

  it('Snapshot', () => {
    const { asFragment } = render(<BaseButton label='Primary' buttonType='button' onClick={() => jest.fn()} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render label with icon', () => {
    // @ts-ignore
    const { container } = render(<WithIcon {...WithIcon.args} />);
    expect(container.firstChild).toHaveClass('base-btn-with-icon');
  });

  it('should render label with icon', () => {
    // @ts-ignore
    const { container } = render(<IconButton {...IconButton.args} />);
    expect(container.firstChild).toHaveClass('base-icon-btn');
  });
});
