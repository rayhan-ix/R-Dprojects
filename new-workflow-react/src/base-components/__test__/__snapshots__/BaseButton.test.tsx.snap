// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Testing Snapshot 1`] = `
<DocumentFragment>
  <button
    class="ant-btn ant-btn-primary base-btn-without-icon"
    type="button"
  >
    <span>
      Primary
    </span>
  </button>
</DocumentFragment>
`;
